test/integration_test.py

```diff
@@ -1099,38 +1096,25 @@ def _get_indexed_bundles(self,
                              filters: Optional[JSON] = None
                              ) -> set[SourcedBundleFQID]:
         indexed_fqids = set()
-        # FIXME: Use `bundles` index for `catalog_complete` subtest
-        #        https://github.com/DataBiosphere/azul/issues/5214
-        hits = self._get_entities(catalog, 'files', filters)
-        if config.is_anvil_enabled(catalog):
-            # Primary bundles may not contain any files, and supplementary
-            # bundles contain only a file and a dataset. We can't use
-            # datasets to find all the indexed bundles because the number of
-            # bundles per dataset often exceeds the inner entity aggregation
-            # limit. Hence, we need to collect bundles separately for files
-            # and biosamples to cover supplementary and primary bundles,
-            # respectively.
-            hits.extend(self._get_entities(catalog, 'biosamples', filters))
+        hits = self._get_entities(catalog, 'bundles', filters)
         for hit in hits:
             source = one(hit['sources'])
-            for bundle in hit.get('bundles', ()):
-                bundle_fqid = SourcedBundleFQIDJSON(uuid=bundle['bundleUuid'],
-                                                    version=bundle['bundleVersion'],
-                                                    source=SourceJSON(id=source['sourceId'],
-                                                                      spec=source['sourceSpec']))
-                if config.is_anvil_enabled(catalog):
-                    is_supplementary = only(set(chain.from_iterable(
-                        always_iterable(file['is_supplementary'])
-                        for file in hit['files']
-                    )), default=False)
-                    bundle_fqid = cast(TDRAnvilBundleFQIDJSON, bundle_fqid)
-                    bundle_fqid['entity_type'] = (
-                        BundleEntityType.supplementary
-                        if is_supplementary else
-                        BundleEntityType.primary
-                    ).value
-                bundle_fqid = self.repository_plugin(catalog).resolve_bundle(bundle_fqid)
-                indexed_fqids.add(bundle_fqid)
+            bundle = one(hit['bundles'])
+            bundle_fqid = SourcedBundleFQIDJSON(uuid=bundle['bundleUuid'],
+                                                version=bundle['bundleVersion'],
+                                                source=SourceJSON(id=source['sourceId'],
+                                                                  spec=source['sourceSpec']))
+            if config.is_anvil_enabled(catalog):
+                if len(hit['biosamples']) > 0:
+                    entity_type = BundleEntityType.primary
```

Comment:
It would be good to represent the bundle entity type in the bundle inner entity, at least on the bundles endpoint, or everywhere if that's easy. Lastly, could you check what the accumulation limit is on bundle inner entities? I have the feeling that's what's freezing up the Swagger UI for the `/index/datasets` endpoint. It may be 1000 which seems excessive now.

Reply:
> check what the accumulation limit is on bundle inner entities?

100.

Reply:
> It would be good to represent the bundle entity type in the bundle inner entity, at least on the bundles endpoint, or everywhere if that's easy.

It is not going to be easy, for any index, including `bundles`. This is because "bundle" inner entities are derived from contributions' _coordinates_ instead of their contents. This happens [deep into the indexer code](https://github.com/DataBiosphere/azul/blob/225b31d8c2f3e7996680c59e24cfd1b7f3944ade/src/azul/indexer/document.py#L1358) where none of the classes are pluginified. Additionally, [this comment](https://github.com/DataBiosphere/azul/blob/225b31d8c2f3e7996680c59e24cfd1b7f3944ade/src/azul/indexer/document.py#L521) claims that we can't add fields from `BundleFQID` subclasses to the index, although I don't entirely understand its rationale.

Reply:
Decided at PL to add additional explanations to the integration test code that infers the bundle type.